osrf/vrx-docker
# New yaml for wildlife encounter task

* This PR adds a wildlife.yaml file that generates three practice worlds similar to the 3 stored in the vrx repository under `vrx_gazebo/worlds/2022_practice`.
* It requires the new file `wildlife.xacro` in the vrx repository, so this PR must be tested in conjunction with [PR 390](https://github.com/osrf/vrx/pull/390) of that repository.

## How to test:
These instructions assume you have a workspace at `~/vrx_ws` with both vrx and vrx-docker installed, and have checked out the `M1chaelM/wildlife_xacro` branch in vrx.
* First, switch to the root of the `vrx-docker` repository.  
  ```
  cd ~/vrx_ws/src/vrx-docker
  ```
* Set some variables for convenience:
  ```
  TASK=wildlife
  TRIAL=0
  ```
* Generate the trial worlds for the task
  ```
  ./prepare_task_trials.bash $TASK
  ```
* Verify that 3 worlds are generated without errors (except the expected `REQUIRED process [world_gen-2] has died!` message).
* Now for each trial, 0-2
    * Test that the generated world runs:
      ```
      roslaunch vrx_gazebo vrx.launch world:=`pwd`/generated/task_generated/${TASK}/worlds/${TASK}${TRIAL}.world verbose:=true wamv_locked:=true
      ```
    * Check that the generated world matches the corresponding world in `vrx/vrx_gazebo/worlds/2022_practice` in all important details. The way I do this is by looking at the diff with vimdiff. For example:
      ```
      cd ~/vrx_ws/src/
      vimdiff vrx-docker/generated/task_generated/${TASK}/worlds/${TASK}${TRIAL}.world 
      vrx/vrx_gazebo/worlds/2022_practice/${TASK}${TRIAL}.world
      ```
